src/linear_algebra/affine_space.lean

```diff
@@ -15,19 +15,71 @@ open_locale classical
 # Affine spaces
 
 This file defines affine spaces (over modules) and subspaces, affine
-maps, affine combinations of points, and the affine span of a set of
-points.
+maps, affine combinations of points, affinely independent families of
+points, and the affine span of a set of points.
+
+## Main definitions
+
+* `affine_space k V P` is an abbreviation for `add_torsor V P` in the
+  case of `module k V`.  Definitions and results not depending on the
+  `module` structure appear in `algebra.add_torsor` instead of here;
+  that includes the instance of an `add_group` as an `add_torsor` over
+  itself, which thus gives a `module` as an `affine_space` over
+  itself.  Definitions of affine spaces vary as to whether a space
+  with no points is permitted; here, we require a nonempty type of
+  points (via the definition of torsors requiring a nonempty type).
+  Affine spaces are defined over any module, with stronger type class
+  requirements on `k` being used for individual lemmas where needed.
+* `affine_subspace k V P` is the type of affine subspaces.  Unlike
+  affine spaces, affine subspaces are allowed to be empty, and lemmas
+  that do not apply to empty affine subspaces have `nonempty`
+  hypotheses.  There is a `complete_lattice` structure on affine
+  subspaces.
```

Comment:
As far as I can tell, affine subspaces don't have an `affine_space` instance, correct? I would like to see that noted here (if you expect they will not get one) or in the implementation notes (if you are planning to define this instance later).

Reply:
Added some discussion to the implementation notes of specific definitions that are missing, including this one.